TheraLang.Tests/Services/UserManagmentServiceTests.cs

```diff
@@ -1,10 +1,230 @@
-﻿using System;
+﻿using Common.Helpers.PasswordHelper;
+using FluentAssertions;
+using MockQueryable.Moq;
+using Moq;
+using System;
 using System.Collections.Generic;
-using System.Text;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using TheraLang.BLL.Interfaces;
+using TheraLang.BLL.Services;
+using TheraLang.DAL.Entities;
+using TheraLang.DAL.Repository;
+using TheraLang.DAL.UnitOfWork;
+using TheraLang.Tests.DataBuilders.ResourcesBuilders;
+using Xunit;
 
 namespace TheraLang.Tests.Services
 {
-    class UserManagmentServiceTestscs
+    public class UserManagmentServiceTests
     {
+        private readonly Mock<IUnitOfWork> _unitOfWorkMock;
+        private readonly Mock<IFileService> _fileService;
+        private readonly Mock<IConfirmationService> _confirmationService;
+        private readonly UserManagementService _userService;
+        private readonly List<User> _testUsersDb;
+        private readonly List<Role> _testRolesDb;
+        private readonly List<UserDetails> _testUserDetailsDb;
+        private readonly List<UserConfirmation> _testUserConfirmationsDb;
+         
+        public UserManagmentServiceTests()
+        {
+            _testUsersDb = GetUsersTestData().ToList();
+            _testRolesDb = GetRolesTestData().ToList();
+            _testUserDetailsDb = GetUserDetailsTestData().ToList();
+            _testUserConfirmationsDb = GetUserConfirmationsTestData().ToList();
+
+            var userRepoMock = new Mock<IRepository<User>>();
+            var userDetailsRepoMock = new Mock<IRepository<UserDetails>>();
+            var roleRepoMock = new Mock<IRepository<Role>>();
+            var userConfirmationRepoMock = new Mock<IRepository<UserConfirmation>>();
+            userRepoMock.Setup(r => r.GetAll()).Returns(_testUsersDb.AsQueryable().BuildMock().Object);
```

Comment:
i guess we can move setup to setup_method

Reply:
done. I created generic repository setup. So we can use it in all project, as we have same setup everywhere